internal/controller/volsync/vshandler.go

```diff
@@ -1236,29 +1303,102 @@ func (v *VSHandler) DeleteRS(pvcName string, pvcNamespace string) error {
 	return nil
 }
 
-//nolint:nestif
-func (v *VSHandler) DeleteRD(pvcName string, pvcNamespace string) error {
-	// Remove a ReplicationDestination by name that is owned (by parent vrg owner)
+func (v *VSHandler) removeRDAsOwnerFromPVC(
+	rd *volsyncv1alpha1.ReplicationDestination,
+	pvcName, pvcNamespace string,
+) error {
+	// Removing the ownership should and can occur only if the resources in question reside in the same namespace.
+	// Otherwise, we exit early
+	if rd.Namespace != pvcNamespace {
+		return nil
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	key := types.NamespacedName{
+		Namespace: pvcNamespace,
+		Name:      pvcName,
+	}
+
+	if err := v.client.Get(v.ctx, key, pvc); err != nil {
+		v.log.Error(err, "Failed to get PVC for disowning", "pvc", key)
```

Comment:
If the pvc does not exist we should not fail. Otherwise we want to return an error with some context without logging the error.